subprojects/core/src/main/java/org/gradle/api/tasks/AbstractCopyTask.java

```diff
@@ -169,7 +169,7 @@ protected CopyActionExecuter createCopyActionExecuter() {
      * @return The source files. Never returns null.
      */
     @Internal
-    @ToBeReplacedByLazyProperty
+    @NotToBeReplacedByLazyProperty(because = "Read-only nested like property")
     public FileCollection getSource() {
```

Comment:
I don't think there is a plan for this particular property. 

For that one in `Sign`: we discussed that we will deprecate it, so that is why it has `willBeDeprecated = true`